mirrord/cli/src/internal_proxy.rs

```diff
@@ -146,6 +146,29 @@ pub(crate) async fn proxy() -> Result<()> {
     Ok(())
 }
 
+/// Connect and send ping - this is useful when working using k8s
+/// port forward since it only creates the connection after
+/// sending the first message
+async fn connect_and_ping(
+    config: &LayerConfig,
+) -> Result<(mpsc::Sender<ClientMessage>, mpsc::Receiver<DaemonMessage>)> {
+    let (mut sender, mut receiver) = connect(config).await?;
+    ping(&mut sender, &mut receiver).await?;
+    Ok((sender, receiver))
+}
+
+/// Sends a ping the connection and expects a pong.
+async fn ping(
+    sender: &mut mpsc::Sender<ClientMessage>,
+    receiver: &mut mpsc::Receiver<DaemonMessage>,
+) -> Result<(), InternalProxyError> {
+    sender.send(ClientMessage::Ping).await?;
+    match receiver.recv().await {
+        Some(DaemonMessage::Pong) => Ok(()),
+        _ => Err(InternalProxyError::AgentClosedConnection),
```

Comment:
Maybe should be like CliError::InvalidMessage to be more indicative that it didn't get desired pong back